Makefile

```diff
@@ -75,22 +75,23 @@ CXXFLAGS += $(ALSA_CFLAGS)
 LDFLAGS += $(ALSA_LDFLAGS)
 endif
 
-ASSISTANT_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_audio.o ./src/run_assistant_text.o
-ASSISTANT_AUDIO_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_audio.o
-ASSISTANT_FILE_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_file.o
-ASSISTANT_TEXT_O = ./src/json_util.o ./src/base64_encode.o ./src/run_assistant_text.o
+CORE_SRCS = ./src/base64_encode.cc ./src/json_util.cc
+
+ASSISTANT_O       = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_audio.o ./src/run_assistant_text.o
+ASSISTANT_AUDIO_O = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_audio.o
+ASSISTANT_FILE_O  = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_file.o
+ASSISTANT_TEXT_O  = $(CORE_SRCS:.cc=.o) \
+                    ./src/run_assistant_text.o
 
 .PHONY: all
-all: run_assistant
+all: run_assistant_audio run_assistant_file run_assistant_text
```

Comment:
I would run this with `make all`?

I am still keeping the `run_assistant` logic for [`tests/build.sh`](https://github.com/googlesamples/assistant-sdk-cpp/blob/master/tests/build.sh#L50) which I run on a CI server before submitting a change. That way I can make sure they all compile with a single command.

If we were to change this command, we'd need to update `build.sh` as well.

Reply:
Understood. I'd add it to .PHONY tough.